homework/src/index.js

```diff
@@ -36,11 +36,66 @@
       if (err) {
         createAndAppend('div', root, { html: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
+        const topDiv = createAndAppend('div', root, { class: 'topDiv' });
+        const header = createAndAppend('header', topDiv, { class: 'heading' });
+        createAndAppend('h1', header, { html: 'HYF Repositories ', class: 'h1-text' });
+        const selectPanel = createAndAppend('select', header, { class: 'select-panel' });
+        data.sort(function (a, b) { return a.name.localeCompare(b.name); });
+        for (let i = 0; i < data.length; i++) {
+          createAndAppend('option', selectPanel, { value: i, html: data[i].name });
+        }
+        const LeftDiv = createAndAppend('div', root, { class: 'leftDiv' });
+        const rightDiv = createAndAppend('div', root, { class: 'rightDiv' });
+        selectPanel.addEventListener('change', () => {
+          const value = data[selectPanel.selectedIndex].url;
+          fetchJSON(value, (err, data) => {
+            renderReposInfo(LeftDiv, data, value);
+          });
+          const contributors = data[selectPanel.selectedIndex].contributors_url;
+          fetchJSON(contributors, (err, data) => {
+            renderContributorsInfo(rightDiv, data, contributors);
+          });
+        });
+        fetchJSON(data[0].url, (err, data) => {
+          renderReposInfo(LeftDiv, data);
+        });
+        fetchJSON(data[0].contributors_url, (err, data) => {
+          renderContributorsInfo(rightDiv, data);
+        });
       }
     });
   }
 
+  function addRow(label, value, tbody) {
+    const row = createAndAppend('tr', tbody);
+    createAndAppend('td', row, { html: label + ':  ', class: 'label' });
+    createAndAppend('td', row, { html: value, class: 'repo-name', target: '_blank' });
+  }
+
+  function renderReposInfo(leftDiv, data) {
+    leftDiv.innerHTML = '';
+    const table = createAndAppend('table', leftDiv, { class: 'table-item' });
+    const tbody = createAndAppend('tbody', table);
+    addRow('Repository', data.name.link(data.html_url), tbody);
+    addRow('Description', data.description, tbody);
+    addRow('Forks', data.forks, tbody);
+    addRow('Updated', data.updated_at, tbody);
+  }
+
+  function renderContributorsInfo(rightDiv, data) {
+    rightDiv.innerHTML = '';
+    const rightTable = createAndAppend('div', rightDiv, { class: 'right-table' });
+    createAndAppend('p', rightTable, { html: 'Contributions', class: 'cont-title' });
+    const ul = createAndAppend('ul', rightTable, { class: 'contributors-list' });
+    data.forEach(contributor => {
+      const contributorLink = createAndAppend('a', ul, { href: (`${contributor.html_url}`), target: '_blank', class: 'c-link' });
```

Comment:
You usually have an unordered list element `<ul>`, with inside that a number of list items `<li>` which can contain a link `<a>`, which can contain an image `<img>` and text `<span>`. 

So you would generate something like this: 
```
<ul>
    <li>
        <a href="#">
            <img src="http://some.source/">
            <span>Some text</span>
        </a>
    </li>

    <li>
        <a href="#other">
            <img src="http://some.orher.source/">
            <span>Some other text</span>
        </a>
    </li>
</ul>
```

You now have: 
```
<ul>
    <a href="#">
        <li>
            <img src="http://some.source/">
            <span>Some text</span>
        </li>
    </a>

    <a href="#other">
        <li>
            <img src="http://some.orher.source/">
            <span>Some other text</span>
        </li>
    </a>
</ul>
```